code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -252,23 +252,40 @@
 	OpenFire()
 
 /mob/living/simple_animal/hostile/asteroid/hivelord/death(gibbed)
-	new /obj/item/asteroid/hivelord_core(src.loc)
+	new /obj/item/organ/internal/hivelord_core(src.loc)
 	mouse_opacity = 1
 	..(gibbed)
 
-/obj/item/asteroid/hivelord_core
+/obj/item/organ/internal/hivelord_core
 	name = "hivelord remains"
 	desc = "All that remains of a hivelord, it seems to be what allows it to break pieces of itself off without being hurt... its healing properties will soon become inert if not used quickly."
-	icon = 'icons/obj/food/food.dmi'
-	icon_state = "boiledrorocore"
+	icon_state = "roro core 2"
+	slot = "hivecore"
 	var/inert = 0
 
-/obj/item/asteroid/hivelord_core/New()
+/obj/item/organ/internal/hivelord_core/New()
+	..()
 	spawn(1200)
```

Comment:
you're going to have to change this if you want people to actually implant it; basically miners will have only a mere 2 minutes to get this thing implanted before it's literally worthless.